pubnub-core/src/subscription/message_destinations.rs

```diff
@@ -0,0 +1,106 @@
+use crate::data::message::{self, Message};
+use crate::data::pubsub;
+
+#[derive(Debug)]
+pub(super) struct MessageDestinations<'a> {
+    message: &'a Message,
+    route: bool,
+    channel: bool,
+}
+
+impl<'a> MessageDestinations<'a> {
+    pub fn new(message: &'a Message) -> Self {
+        Self {
+            message,
+            route: false,
+            channel: false,
+        }
+    }
+
+    fn route_destination(&self) -> Option<pubsub::SubscribeTo> {
+        match self.message.route {
+            Some(message::Route::ChannelGroup(ref val)) => {
+                Some(pubsub::SubscribeTo::ChannelGroup(val.clone()))
+            }
+            Some(message::Route::ChannelWildcard(ref val)) => {
+                Some(pubsub::SubscribeTo::ChannelWildcard(val.clone()))
+            }
```

Comment:
Cloning here is the only thing in this PR I'm not too fond of. AFAICT, these are used by the `Iterator` impl, but that could just borrow the names, and we could add an `Into<Iterator>` impl to transfer ownership of the strings where necessary.

But we can save the `clone` audit for later!